### PR TITLE
Re-implements network check with vendored `psutil`

### PR DESCRIPTION
**Pull request by luiscape**
Adds the same logic from [this PR](https://github.com/modal-labs/modal-client/pull/1347) but with vendored `psutil` so we don't have to install an additional package.

### CODE REVIEW

**Comment by ekzhang** [comment]
Can you include the copyright notice?

https://github.com/giampaolo/psutil?tab=BSD-3-Clause-1-ov-file

File: vendor/psutil.py [hunk 1]
```diff
@@ -0,0 +1,356 @@
+# Copyright Modal Labs 2022
```
Comment (mwaskom) [inline vendor/psutil.py:1]: Wrong!

**Comment by mwaskom** [comment]
Also can we prefix the location with an underscore (e.g. `_vendor` so that it doesn't get confused for public API?

**Comment by mwaskom** [comment]
What is our approach to tests going to be with vendored code? Are we just going to assume that our existing unit tests cover all the behaviors that we need?

**Comment by mwaskom** [comment]
It also feels like we want some way of tracking what version the vendored code is sourced from. How will we know when we need to update it, e.g. if there is an important bug in the source repository?

**Comment by ekzhang** [comment]
While these are reasonable comments, I don't think they are relevant to this PR, as the use of `psutil` here is just a particular function that just reads and parses `/proc`. We can revisit it if we need more functionality.

**Comment by mwaskom** [comment]
Now is probably the only time we will know what version this corresponds to though :)

**Comment by ekzhang** [comment]
It shouldn't matter what version it corresponds to — barring any changes to the POSIX API, it should not need updates.

**Comment by luiscape** [comment]
> Are we just going to assume that our existing unit tests cover all the behaviors that we need?

This should be the case, yes. The [tests I added](https://github.com/modal-labs/modal-client/pull/1357/files#diff-91e91e76633aefa71ffbbd8080e1176cc7f61a2d01e3d387cb1f05fc058a4f40R915) test our interface and assumes the vendored code works as it should. That's is acceptable imo.

> It also feels like we want some way of tracking what version the vendored code is sourced from.

I'll add a note to comments if only to point to the right license / code version. I agree with Eric that anything more is overkill.

**Pull request closed without merging**